### DISCUSSION TITLE
Does clangd support visual studio solution(.sln)  build system?

### DISCUSSION THREAD

**Discussion by bigdogs** [clangd/clangd#1161] [Q&A]
Hi,

 I found that `clangd` can't find my definition of c++ virtual function.   when I click the virtual function with base class instance, it always 
 jump to the interface of the base class. `clangd` seems knowns nothing about where is the definition :(

  I'm not sure if this is because .sln solution(vs2019 build system) is not supported.

**Comment by HighCommander4** [comment]
It's not supported out of the box; clangd needs a `compile_commands.json` file to understand what files are part of the project and how to parse them.

Please see https://clangd.llvm.org/installation#project-setup for some options for how to prepare such a file.

**Comment by Yurii-huang** [comment]
Specific How the .sln project is implemented is unknown. 
If you are interested, you can check compile_commands.json, which contains some compilation parameters required for construction, including include path, because .sln project can support dynamic addition of header files and macros. 
I speculate that it does a deeper encapsulation of the compiled database.

Maybe Microsoft's official website can give you the corresponding answer.

**Comment by bigdogs** [comment]
Thanks,  although this is not the perfect solution, "compile_command.json" is works

**Comment by sakgoyal** [comment]
https://marketplace.visualstudio.com/items?itemName=caphyon.ClangPowerTools
this will require VS. but you can create a compile_commands file from this.
